### PR TITLE
virsh_edit: support to use vi editor for editing vmxml

### PR DESCRIPTION
**Pull request by balamuruhans**
virsh edit test cases uses vi editor regex to edit vmxml
if editor is not configured or if virsh uses other editor
like nano, all the tests fails.

Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by balamuruhans** [comment]
**Before Fix**
virsh was using nano to edit vmxml and regex dint work

```
2017-01-19 12:57:36,950 libvirt          L2818 INFO | Trying to edit xml with cmd [":%s /<topology .*\\/>/<topology sockets='2' cores='4' threads='8'\\/>", ':%s /[0-9]*<\\/vcpu>/64<\\/vcpu>']
2017-01-19 12:57:47,376 libvirt          L2831 ERROR| Error occurred: Login timeout expired    (output: 'root@powerkvm2-lp1:~#                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 [ Reading File ][ Read 52 lines  GNU nano 2.5.3          File: /tmp/virshCixV5a.^X Exit      ^R Read File ^\\ Replace   ^U Uncut Text^T To Spell  ^_ Go To LineM\x07')
2017-01-19 12:57:47,376 virsh_edit       L0113 INFO | False
2017-01-19 12:57:47,958 virsh            L1355 DEBUG| Undefine VM virt-tests-vm1-bala
2017-01-19 12:57:47,980 virsh            L1339 DEBUG| Define VM from /var/tmp/xml_utils_temp_tobI7Y.xml
2017-01-19 12:57:48,084 libvirt_vm       L1832 WARNI| Requested MAC address release from persistent vm virt-tests-vm1-bala. Ignoring.
2017-01-19 12:57:48,176 virsh            L1355 DEBUG| Undefine VM virt-tests-vm1-bala
2017-01-19 12:57:48,201 virsh            L1339 DEBUG| Define VM from /var/tmp/xml_utils_temp_tobI7Y.xml
2017-01-19 12:57:48,318 qemu_storage     L0489 DEBUG| Checking image file /var/lib/libvirt/images/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/data/avocado-vt/images/ubuntu-17.04-ppc64le.qcow2
2017-01-19 12:57:48,600 process          L0368 INFO | Running 'true'
2017-01-19 12:57:48,603 process          L0482 INFO | Command 'true' finished with 0 after 0.000396966934204s
2017-01-19 12:57:48,603 process          L0368 INFO | Running 'ps -o comm 1'
2017-01-19 12:57:48,677 process          L0462 DEBUG| [stdout] COMMAND
2017-01-19 12:57:48,677 process          L0482 INFO | Command 'ps -o comm 1' finished with 0 after 0.0709900856018s
2017-01-19 12:57:48,677 process          L0462 DEBUG| [stdout] systemd
2017-01-19 12:57:48,812 stacktrace       L0038 ERROR|
2017-01-19 12:57:48,812 stacktrace       L0041 ERROR| Reproduced traceback from: /usr/local/lib/python2.7/dist-packages/avocado_plugins_vt-44.0-py2.7.egg/avocado_vt/test.py:420
2017-01-19 12:57:48,812 stacktrace       L0044 ERROR| Traceback (most recent call last):
2017-01-19 12:57:48,813 stacktrace       L0044 ERROR|   File "/usr/local/lib/python2.7/dist-packages/avocado_plugins_vt-44.0-py2.7.egg/avocado_vt/test.py", line 196, in runTest
2017-01-19 12:57:48,813 stacktrace       L0044 ERROR|     self._runTest()
2017-01-19 12:57:48,813 stacktrace       L0044 ERROR|   File "/usr/local/lib/python2.7/dist-packages/avocado_plugins_vt-44.0-py2.7.egg/avocado_vt/test.py", line 337, in _runTest
2017-01-19 12:57:48,813 stacktrace       L0044 ERROR|     run_func(self, params, env)
2017-01-19 12:57:48,813 stacktrace       L0044 ERROR|   File "/var/lib/libvirt/images/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/data/avocado-vt/test-providers.d/downloads/powerkvm-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_edit.py", line 284, in run
2017-01-19 12:57:48,813 stacktrace       L0044 ERROR|     raise exceptions.TestFail("Run failed with right command")
2017-01-19 12:57:48,813 stacktrace       L0044 ERROR| TestFail: Run failed with right command
2017-01-19 12:57:48,813 stacktrace       L0045 ERROR|
2017-01-19 12:57:48,813 test             L0620 ERROR| FAIL 1-guest_lifecycle.edit.qemu.qcow2.scsi.smp2.virtio_net.Ubuntu.17.04.ppc64le.powerkvm-libvirt.virsh.edit.positive_test.id_option.edit_vcpu -> TestFail: Run failed with right command
```

**Comment by balamuruhans** [comment]
**After Fix**
```
# avocado run --vt-type libvirt --vt-config /var/lib/libvirt/images/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/data/avocado-vt/backends/libvirt/cfg/lifecycle.cfg --vt-only-filter "scsi qcow2 Ubuntu.17.04.ppc64le"
JOB ID     : 3c57901482a301aa0b757d01d3e3206130712c22
JOB LOG    : /var/lib/libvirt/images/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/results/job-2017-01-19T14.42-3c57901/job.log
TESTS      : 1
 (1/1) guest_lifecycle.edit.qemu.qcow2.scsi.smp2.virtio_net.Ubuntu.17.04.ppc64le.powerkvm-libvirt.virsh.edit.positive_test.id_option.edit_vcpu: PASS (5.51 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0
TESTS TIME : 5.51 s
JOB HTML   : /var/lib/libvirt/images/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/results/job-2017-01-19T14.42-3c57901/html/results.html
```

**Comment by balamuruhans** [comment]
@waynesun09 @will-Do please help me reviewing this PR

File: libvirt/tests/src/virsh_cmd/domain/virsh_edit.py [hunk 1]
```diff
@@ -31,6 +34,37 @@ def run(test, params, env):
...
+            subprocess.check_output(['bash', '-c', "source %s" % bashrc_file])
+    except process.CmdError:
+        logging.debug("%s file doesn't exist" % bashrc_file)
+        try:
+            process.system("echo %s > %s" % (editor_cfg,
+                                             bashrc_file), shell=True)
+            subprocess.check_output(['bash', '-c', "source %s" % bashrc_file])
+        except Exception, info:
+            raise exceptions.TestSkipError("Test requires vi editor as "
+                                           "default - %s" % info)
```
Comment (will-Do) [inline libvirt/tests/src/virsh_cmd/domain/virsh_edit.py]: From my point of view, just update os.env should be enough

**Comment by iccaszhulili** [comment]
rh-caselink retest please

**Comment by sathnaga** [comment]
@balamuruhans needs rebase

**Comment by balamuruhans** [comment]
This change will not be necessary as suggested by @will-Do, so let's close the PR.

**Pull request closed without merging**